expression/generator/time_vec.go

```diff
@@ -440,15 +440,32 @@ func (b *{{.SigName}}) vecEvalTime(input *chunk.Chunk, result *chunk.Column) err
 		return nil
 	}
 
+	intervalBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(intervalBuf)
+	if err := b.vecGetIntervalFrom{{.TypeB.ETName}}(&b.baseBuiltinFunc, input, unit, intervalBuf); err != nil {
+		return err
+	}
+
 	{{ if eq .TypeA.TypeName "Duration" }}
-		durationBuf, err := b.bufAllocator.get(types.ETDuration, n)
-		if err != nil {
+		result.ResizeGoDuration(n, false)
+		if err := b.args[0].VecEvalDuration(b.ctx, input, result); err != nil {
 			return err
 		}
-		defer b.bufAllocator.put(durationBuf)
-		if err := b.args[0].VecEvalDuration(b.ctx, input, durationBuf); err != nil {
+
+		result.MergeNulls(intervalBuf)
+		resDurations := result.GoDurations()
+		iterDuration := types.Duration{Fsp: types.MaxFsp}
+	{{ else if eq .TypeA.TypeName "Time" }}
+		result.ResizeTime(n, false)
+		if err := b.vecGetDateFromDatetime(&b.baseBuiltinFunc, input, unit, result); err != nil {
 			return err
 		}
+
+		result.MergeNulls(intervalBuf)
+		resDates := result.Times()
 	{{ else }}
```

Comment:
When will this `else`  be hit?

Reply:
`dateBuf` can also be reused in this section, I think we can treat this section as well as `eq .TypeA.TypeName "Time"`

Reply:
Yes, it's fixed now.

Reply:
Did you forget to commit time_vec.go?

Reply:
Sorry...